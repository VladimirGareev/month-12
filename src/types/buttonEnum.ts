export enum Action {
    Add = "add",
    Delete = "delete",
    AddAtHead = "addAtHead",
    AddAtTail = "addAtTail",
    AddAt = "addAt",
    DeleteFromHead = "deleteFromHead",
    DeleteFromTail = "deleteFromTail",
    DeleteFrom = "deleteFrom",
  }