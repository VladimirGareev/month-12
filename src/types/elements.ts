import { ReactComponentElement } from "react";
import { ElementStates } from "./element-states"

export type TElement = {
    letter: string|number;
    id: number|string;
    state?: ElementStates;
    head?: string|React.ReactElement;
    tail?: string|React.ReactElement;
}

export type TElementBar = {
    number: number;
    id: number;
    state: ElementStates;
}

