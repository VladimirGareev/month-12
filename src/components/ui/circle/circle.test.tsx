import {cleanup, render, screen, fireEvent, within} from '@testing-library/react'
import { Circle } from './circle';
import { ElementStates } from '../../../types/element-states';

describe ("circle component testing", ()=>{
    beforeEach(cleanup);
 it("Circle отрисовывается корректно без буквы", ()=> {
        render (<Circle />);
        const circle = screen.getByTestId("circle");
        expect(circle).toHaveTextContent('');
        expect(circle).toMatchSnapshot();
    })

    it("Circle отрисовывается корректно c буквами", ()=> {
        render (<Circle letter="ABCD"/>);
        const circle = screen.getByTestId("circle");
        expect(circle).toHaveTextContent('ABCD');
        expect(circle).toMatchSnapshot();
    })

    it("Circle отрисовывается корректно c текстом в HEAD", ()=> {
        render (<Circle head="ABCD"/>);
        const head = screen.getByTestId("head");
        expect(head).toHaveTextContent('ABCD');
        expect(head).toMatchSnapshot();
    })
    
    it("Circle отрисовывается корректно c элементом в HEAD", ()=> {
        render (<Circle head={<Circle isSmall letter="ABCD"/>}/>);
        const heads = screen.getAllByTestId("head");
        const circle = within(heads[0]).getByTestId("circle")
        expect(circle).toBeInTheDocument();
        expect(circle).toHaveTextContent('ABCD');
        expect(heads[0]).toMatchSnapshot();
    })
    it("Circle отрисовывается корректно c текстом в TAIL", ()=> {
        render (<Circle tail="ABCD"/>);
        const tail = screen.getByTestId("tail");
        expect(tail).toHaveTextContent('ABCD');
        expect(tail).toMatchSnapshot();
    })
    
    it("Circle отрисовывается корректно c элементом в TAIL", ()=> {
        render (<Circle tail={<Circle isSmall letter="ABCD"/>}/>);
        const tails = screen.getAllByTestId("tail");
        const circle = within(tails[0]).getByTestId("circle")
        expect(circle).toBeInTheDocument();
        expect(circle).toHaveTextContent('ABCD');
        expect(tails[0]).toMatchSnapshot();
    })

    it("Circle отрисовывается корректно c Index", ()=> {
        render (<Circle index={1} />);
        const circle = screen.getByTestId("circle");
        const index = within(circle).getByTestId("index")
        expect(index).toBeInTheDocument();
        expect(index).toHaveTextContent('1');
        expect(circle).toMatchSnapshot();
    })

    it("Circle отрисовывается корректно пропсом isSmall", ()=> {
        render (<Circle isSmall={true} />);
        const circle = screen.getByTestId("circle");
        const small = screen.getByTestId("small");
        expect(circle).toBeInTheDocument();
        expect(small).toHaveClass("small");
        expect(circle).toMatchSnapshot();
    })
    it("Circle отрисовывается корректно в состоянии default", ()=> {
        render (<Circle state={ElementStates.Default} />);
        const circle = screen.getByTestId("circle");
        const small = screen.getByTestId("small");
        expect(circle).toBeInTheDocument();
        expect(small).toHaveClass("default");
        expect(circle).toMatchSnapshot();
    })
    it("Circle отрисовывается корректно в состоянии changing", ()=> {
        render (<Circle state={ElementStates.Changing} />);
        const circle = screen.getByTestId("circle");
        const small = screen.getByTestId("small");
        expect(circle).toBeInTheDocument();
        expect(small).toHaveClass("changing");
        expect(circle).toMatchSnapshot();
    })

    it("Circle отрисовывается корректно в состоянии modified", ()=> {
        render (<Circle state={ElementStates.Modified} />);
        const circle = screen.getByTestId("circle");
        const small = screen.getByTestId("small");
        expect(circle).toBeInTheDocument();
        expect(small).toHaveClass("modified");
        expect(circle).toMatchSnapshot();
    })
})