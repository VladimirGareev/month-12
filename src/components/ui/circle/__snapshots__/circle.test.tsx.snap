// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`circle component testing Circle отрисовывается корректно c Index 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    class="circle   default"
    data-testid="small"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  >
    1
  </p>
  <div
    class="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-testid="tail"
  />
</div>
`;

exports[`circle component testing Circle отрисовывается корректно c буквами 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    class="circle   default"
    data-testid="small"
  >
    <p
      class="text text_type_circle text_color_input letter"
    >
      ABCD
    </p>
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`circle component testing Circle отрисовывается корректно c текстом в HEAD 1`] = `
<div
  class="text text_type_input text_color_input mb-4 absolute head string"
  data-testid="head"
>
  ABCD
</div>
`;

exports[`circle component testing Circle отрисовывается корректно c текстом в TAIL 1`] = `
<div
  class="text text_type_input text_color_input mt-4 absolute tail30 string"
  data-testid="tail"
>
  ABCD
</div>
`;

exports[`circle component testing Circle отрисовывается корректно c элементом в HEAD 1`] = `
<div
  class="text text_type_input text_color_input mb-4 absolute head element"
  data-testid="head"
>
  <div
    class="content "
    data-testid="circle"
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="head"
    />
    <div
      class="circle  small default"
      data-testid="small"
    >
      <p
        class="text text_type_circle text_color_input letter"
      >
        ABCD
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
      data-testid="index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="tail"
    />
  </div>
</div>
`;

exports[`circle component testing Circle отрисовывается корректно c элементом в TAIL 1`] = `
<div
  class="text text_type_input text_color_input mt-4 absolute tail30 element"
  data-testid="tail"
>
  <div
    class="content "
    data-testid="circle"
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="head"
    />
    <div
      class="circle  small default"
      data-testid="small"
    >
      <p
        class="text text_type_circle text_color_input letter"
      >
        ABCD
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
      data-testid="index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="tail"
    />
  </div>
</div>
`;

exports[`circle component testing Circle отрисовывается корректно без буквы 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    class="circle   default"
    data-testid="small"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`circle component testing Circle отрисовывается корректно в состоянии changing 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    class="circle   changing"
    data-testid="small"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`circle component testing Circle отрисовывается корректно в состоянии default 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    class="circle   default"
    data-testid="small"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`circle component testing Circle отрисовывается корректно в состоянии modified 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    class="circle   modified"
    data-testid="small"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`circle component testing Circle отрисовывается корректно пропсом isSmall 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    class="circle  small default"
    data-testid="small"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;
